benchmark/runbench.d

```diff
@@ -87,43 +87,54 @@ void runTests(Config cfg)
         import std.datetime, std.algorithm : min;
         auto sw = StopWatch(AutoStart.yes);
         auto minDur = Duration.max;
+        string minGCProf;
 
-        stdout.writef("R %-16s", bin.baseName.stripExtension);
-        if (cfg.verbose) stdout.writeln();
-        stdout.flush();
+        immutable benchName = bin.baseName.stripExtension;
+
+        void report(string pfx, Duration dur, string gcProf)
+        {
+            auto parts = dur.split!("seconds", "msecs");
+            if (gcProf.length)
+                writefln("%s %-16s %s.%03s s, %s", pfx, benchName, parts.seconds, parts.msecs, gcProf);
+            else
+                writefln("%s %-16s %s.%03s s", pfx, benchName, parts.seconds, parts.msecs);
+        }
 
         auto cmd = bin ~ " " ~ cfg.args;
-        string gcprof;
         foreach (_; 0 .. cfg.repeat)
         {
             sw.reset;
             auto output = runCmd(cmd, cfg.verbose);
             auto dur = cast(Duration)sw.peek;
-            if (cfg.verbose) stdout.write(output);
 
-            if (dur >= minDur) continue;
-            minDur = dur;
+            auto parts = dur.split!("seconds", "msecs");
 
+            if (cfg.verbose) stdout.write(output);
+
+            string gcProf;
             if (exists("gcx.log"))
             {
                 auto tgt = bin.setExtension("gcx.log");
                 rename("gcx.log", tgt);
                 auto lines = File(tgt, "r").byLine()
                     .find!(ln => ln.canFind("GC summary:"));
-                if (!lines.empty) gcprof = lines.front.find("GC summary:")[11..$].idup;
+                if (!lines.empty) gcProf = lines.front.find("GC summary:")[11..$].idup;
             }
             else
             {
                 auto lines = output.splitter(ctRegex!`\r\n|\r|\n`)
                     .find!(ln => ln.startsWith("GC summary:"));
-                if (!lines.empty) gcprof = lines.front[11..$];
+                if (!lines.empty) gcProf = lines.front[11..$];
+            }
+            report("RUN", dur, gcProf);
```

Comment:
I think this should not be displayed in a standard run if you just look at the text output. Maybe print in verbose mode only?